### PR TITLE
TICKET-T-20543: Handle SDK Initialisation failure on app launch

### PR DESCRIPTION
**Pull request by mr-pant**

### CODE REVIEW

File: lib/l10n/app_en.arb [hunk 1]
```diff
@@ -1641,6 +1641,7 @@
   "@routingErrorText": {
     "description": "Routing error message"
   },
+  "sdkInitFailError": "The initialisation of the HERE SDK failed. Please check your code and/or start the native engine with the default options.",
```
Comment (Guru-cm-here) [inline lib/l10n/app_en.arb:1644]: Please add 'description'  to the new string definition.

File: lib/main.dart [hunk 2]
```diff
@@ -57,6 +57,7 @@ void main() {
       Environment.accessKeySecret,
     ),
     onSuccess: () => runApp(MyApp()),
+    onFailure: (_) => runApp(const InitErrorScreen()),
```
Comment (Guru-cm-here) [inline lib/main.dart:60]: Please make the `onSuccess` and `onFailure` parameters non-null and required, as they are mandatory for loading further screens.

Comment (mr-pant) [reply lib/main.dart:60]: The function on its own is just responsible for creating the engine for which only SDKOptions are mandatory hence its kept required, onSuccess or onFailure is optional as they are not needed for creating engine and the handling depends on the use case.

File: lib/l10n/app_en.arb (hunk 1 above)
Comment (mr-pant) [reply lib/l10n/app_en.arb:1644]: Done

File: lib/main.dart (hunk 2 above)
Comment (Guru-cm-here) [reply lib/main.dart:60]: I understand that the primary responsibility of this function is to create the engine, and SDKOptions are mandatory for this purpose. However, considering that the loading of further app screens depends on the success or failure of this operation. These functions are directly tied to the loading of further app screens.
Making them required ensures that the necessary handling is in place for the subsequent screens to load successfully

Comment (mr-pant) [reply lib/main.dart:60]: Making the functions mandatory can only make sure that the function is not null and does not ensure the loading of screens in this case, that handling success/failure is use case specific for which general function is not responsible.

**Review by Guru-cm-here — APPROVED** [review]

**Pull request merged**